tests/jballerina-unit-test/src/test/resources/test-src/statements/compoundassignment/compound_assignment.bal

```diff
@@ -290,3 +290,65 @@ function testCompoundAssignmentAdditionWithRecordAccess() returns int {
     x += (ibm["count"] + arr[0]);
     return x;
 }
+
+function testCompoundAssignmentAdditionArrayElementFunctionInvocation() returns [int, int, int] {
```

Comment:
Can we also add tests where the execution order of the nested indices matter?

For example, for something like
```ballerina
i[foo()][bar()] += 3
```

a test where we validate that foo() is executed first and then bar() ?

Reply:
Fixed